src/test_lib_json/main.cpp

```diff
@@ -3578,6 +3578,19 @@ JSONTEST_FIXTURE_LOCAL(BuilderTest, settings) {
   }
 }
 
+struct BomTest : JsonTest::TestCase {};
+
+JSONTEST_FIXTURE_LOCAL(BomTest, withBom) {
```

Comment:
We probably should have a feature *not* to skipBom, but we can worry about that if/when it comes up.

(There is a JSON parser checker out there that might expect us to fail on BOM, but we already fail on some other rare edge cases, as most JSON parsers do.)

Reply:
Do you mean that we'd better to add extra parameters in the relevant API, to let the caller decide when and where to skip it?

Reply:
Well, I'm saying that your change will probably be acceptable to everyone, but if somebody complains (which is unlikely), then we can add a feature to turn it off.